src/chttpd/src/chttpd_db.erl

```diff
@@ -1592,6 +1676,18 @@ extract_header_rev(Req, ExplicitRev) ->
     end.
 
 
+% cannot partition a system database
+validate_partition_database_create(DbName, Partitioned) ->
+    SystemId = DbName =:= ?l2b(config:get("mem3", "shards_db", "_dbs")) orelse
+        lists:member(DbName, ?SYSTEM_DATABASES),
```

Comment:
This will not work for cases when the database name contain `/`. For example the following would be broken (example is taken from https://github.com/apache/couchdb/pull/1647#issuecomment-427804738 where one of CouchDb users mention his use case).
```
account%2F85%2Fea%2F6075c6c1e266f8512e2233541bdb-201807
```

I think this should use `couch_db:is_systemdb(DbName)` which abstracts that logic (keep in mind that there is an open PR which fixes it a little https://github.com/apache/couchdb/pull/1647).

Reply:
We would need to make `is_systemdb` function public and possibly rename it since we have public `is_system_db` already.